test/TestCriteria.cpp

```diff
@@ -0,0 +1,124 @@
+#include "../src/Criteria.h"
+#include "../src/Criterion.h"
+#include "gtest/gtest.h"
+#include <sstream>
+#include <utility>
+
+TEST(TestCriteria, TestBaseConstructorWithNbAndId) {
+  std::string id = "test";
+  Criteria crits = Criteria(2, id);
```

Comment:
we should be careful here with variables name ("id" here is used for another purpose in the next lines)

Reply:
what do you mean ? 
id is just for the testing.